### PR TITLE
RevEng: Scaffold constructors for DbContext

### LINKED ISSUE #11570: Scaffold tool: add a flag to generate the constructor with DbContextOptions injected
Currently, scaffold existing database tool will generate an `OnConfiguring` method with connection string inside:

```cs
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
{
    #warning To protect potentially sensitive information in your connection string, you should move it out of source code. See http://go.microsoft.com/fwlink/?LinkId=723263 for guidance on storing connection strings.
    optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=Blogging;Trusted_Connection=True;");
}
```

I realized this result is by-design to make the `the result of reverse engineering to be runnable` (https://github.com/aspnet/EntityFrameworkCore/issues/10432), however, shouldn't there be an flag for people who needs an injected `DbContextOptions` instead?

`--inject-dbcontextoptions`

```cs
public BloggingContext(DbContextOptions<BloggingContext> options)
    : base(options)
{ }
```

With this flag, we can re-scaffold the context from existing database without touching the constructor again.

I mean, people who use EF Core in ASP.NET Core MVC app will tend to add the DB context into the service collection using `AddDbContextPool`...

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #11570

### CODE REVIEW

**Comment by ErikEJ** [comment]
Great!

File: src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs [hunk 1]
```diff
@@ -125,6 +126,20 @@ public virtual string WriteCode(IModel model, string @namespace, string contextN
             _sb.AppendLine("}");
         }
 
+        private void GenerateConstructors(string contextName)
+        {
+            _sb.AppendLine($"public {contextName}()")
+                .AppendLine("{}")
```
Comment (bricelam) [inline src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: Separate lines, please

Comment (smitpatel) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]: @bricelam - separate lines for? Can you check output in tests? And put an example of how output should look like?

Comment (bricelam) [reply src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.cs]:
``` csharp
public MyContext()
{
}
```

**Pull request merged**